index.js

```diff
@@ -12,6 +14,11 @@ module.exports = (input, needle, replacement) => {
 	let ret = '';
 	let matchCount = 0;
 	let prevIndex = 0;
+	const startIndex = opts.fromIndex > 0 ? opts.fromIndex : 0;
+
+	if (startIndex > input.length) {
+		return input;
+	}
 
 	while (true) { // eslint-disable-line no-constant-condition
 		const index = input.indexOf(needle, prevIndex);
```

Comment:
`fromIndex` needs be used in the `indexOf` call. The whole point of this option is that it only searches from the set `fromIndex` position.